lib/puppet/util/windows/process.rb

```diff
@@ -9,7 +9,17 @@ module Puppet::Util::Windows::Process
   WAIT_TIMEOUT = 0x102
 
   def execute(command, arguments, stdin, stdout, stderr)
-    Process.create( :command_line => command, :startup_info => {:stdin => stdin, :stdout => stdout, :stderr => stderr}, :close_handles => false )
+    args = {
+      :command_line => command,
+      :startup_info => {:stdin => stdin, :stdout => stdout, :stderr => stderr},
+      :close_handles => false
+    }
+    args[:cwd] = arguments[:cwd] if arguments[:cwd]
+
+    # TODO (PUP-6919): Should we wrap a bad/nonexistent CWD in a separate exception? 
```

Comment:
Whether we can centralized the check or not, I think it'd be good for all platforms to return similar exceptions with similar error messages

Reply:
Agreed. If we can't centralize the `File.directory?` check, I'll go ahead and wrap the syscall error in a more readable error message.